### PR TITLE
Refactoring for `make-hand-service.py`

### PR DESCRIPTION
**Pull request by jtherrmann**
Makes the following changes:
* Logging with `print` statements.
* Use tempfiles where appropriate.
* Add cli options.

### CODE REVIEW

**Comment by jtherrmann** [comment]
I'm currently running the script with:
```
python make_hand_service.py /home/arcgis/jtherrmann/ --rasters-filter='.*Copernicus_DSM_COG_10_N08_00_W01*'
```
But it fails; full log output:
```
CreateFileGDB
CreateMosaicDataset
AddFields
AddRastersToMosaicDataset
CalculateFields
BuildFootprints
Traceback (most recent call last):
  File "make_hand_service.py", line 90, in <module>
    update_boundary='UPDATE_BOUNDARY',
  File "/home/arcgis/miniconda3/envs/arcpy/lib/python3.7/site-packages/arcpy/management.py", line 14697, in BuildFootprints
    raise e
  File "/home/arcgis/miniconda3/envs/arcpy/lib/python3.7/site-packages/arcpy/management.py", line 14694, in BuildFootprints
    retval = convertArcObjectToPythonObject(gp.BuildFootprints_management(*gp_fixargs((in_mosaic_dataset, where_clause, reset_footprint, min_data_value, max_data_value, approx_num_vertices, shrink_distance, maintain_edges, skip_derived_images, update_boundary, request_size, min_region_size, simplification_method, edge_tolerance, max_sliver_size, min_thinness_ratio), True)))
  File "/home/arcgis/miniconda3/envs/arcpy/lib/python3.7/site-packages/arcpy/geoprocessing/_base.py", line 512, in <lambda>
    return lambda *args: val(*gp_fixargs(args, True))
arcgisscripting.ExecuteError: ERROR 999999: Something unexpected caused the tool to fail. Contact Esri Technical Support (http://esriurl.com/support) to Report a Bug, and refer to the error help for potential solutions or workarounds.
No mosaic dataset item selected. This operation cannot be performed.
Failed to execute (BuildFootprints).

Start Time: Wednesday, February 22, 2023 9:16:36 PM
ERROR 999999: Something unexpected caused the tool to fail. Contact Esri Technical Support (http://esriurl.com/support) to Report a Bug, and refer to the error help for potential solutions or workarounds.
No mosaic dataset item selected. This operation cannot be performed.
Failed to execute (BuildFootprints).
Failed at Wednesday, February 22, 2023 9:16:38 PM (Elapsed Time: 2.59 seconds)
```
I wonder if perhaps the regex filter is not matching any rasters? This will be my next line of investigation.

**Comment by asjohnston-asf** [comment]
Drop the `.`s from the filter, I suspect. Per https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/add-rasters-to-mosaic-dataset.htm by default you're doing simple wildcard matching, e.g. `'*Copernicus_DSM_COG_10_N08_00_W01*'. It's only if you put the word `REGEX` in the filter that you get full perl regular expressions.

**Comment by jtherrmann** [comment]
> Drop the `.`s from the filter, I suspect. Per https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/add-rasters-to-mosaic-dataset.htm by default you're doing simple wildcard matching, e.g. `'*Copernicus_DSM_COG_10_N08_00_W01*`. It's only if you put the word `REGEX` in the filter that you get full perl regular expressions.

Ah, I see, I added `REGEX` because I thought you just left that part out of the filter you gave me. So I thought you had just forgotten the `.`.

File: image_services/glo_30_hand/make_hand_service.py [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+    default='.*Copernicus_DSM_COG_10_[NS][0-8]\d_00_[EW]\d\d\d_00_HAND.tif',
+    help=(
+        'Rasters from the glo-30-hand collection will be filtered through this '
+        'regex before they are added to the mosaic dataset.'
+    )
+)
+args = parser.parse_args()
+
 
 dataset_name = 'GLO30_HAND'
```
Comment (asjohnston-asf) [inline image_services/glo_30_hand/make_hand_service.py:20]: There's potential for clobbering an already-existing overview during the `aws s3 cp` step, so we we may want to parameterize `dataset_name` as well.

**Comment by asjohnston-asf** [comment]
I ran it as `python make_hand_service.py --dataset-name GLO30_HAND_ASJ --rasters-filter '*Copernicus_DSM_COG_10_N08_00_W01*' /home/arcgis/asjohnston/gis-services/image_services/glo_30_hand/` and everything worked how I expected. The .sd file published with no issues.

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**